File: src/main/java/mediathek/gui/actions/ToggleBlacklistAction.java
package mediathek.gui.actions;

import mediathek.config.Daten;
import mediathek.gui.messages.BlacklistChangedEvent;
import mediathek.swing.IconUtils;
import mediathek.tool.ApplicationConfiguration;
import mediathek.tool.MessageBus;
import net.engio.mbassy.listener.Handler;
import org.kordamp.ikonli.materialdesign2.MaterialDesignL;
import org.kordamp.ikonli.swing.FontIcon;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;

public class ToggleBlacklistAction extends AbstractAction {
    private final FontIcon enabledIcon;
    private final FontIcon disabledIcon;
    private boolean blacklist_is_on;

    public ToggleBlacklistAction() {
        enabledIcon = IconUtils.windowBarSpecificToolbarIcon(MaterialDesignL.LIST_STATUS);
        disabledIcon = IconUtils.windowBarSpecificToolbarIcon(MaterialDesignL.LIST_STATUS, Color.RED);

        blacklist_is_on = ApplicationConfiguration.getConfiguration().getBoolean(ApplicationConfiguration.BLACKLIST_IS_ON, false);
        setupState();

        MessageBus.getMessageBus().subscribe(this);
    }

    @Handler
    private void handleBlacklistChangedEvent(BlacklistChangedEvent e) {
        //config was changed outside
        SwingUtilities.invokeLater(() -> {
            blacklist_is_on = ApplicationConfiguration.getConfiguration().getBoolean(ApplicationConfiguration.BLACKLIST_IS_ON, false);
            setupState();
        });
    }

    private void setupState() {
        if (blacklist_is_on) {
            putValue(NAME, "Blacklist ausschalten");
            putValue(SHORT_DESCRIPTION, "Blacklist ausschalten");
            putValue(SMALL_ICON, enabledIcon);
        } else {
            putValue(NAME, "Blacklist einschalten");
            putValue(SHORT_DESCRIPTION, "Blacklist einschalten");
            putValue(SMALL_ICON, disabledIcon);
        }
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        blacklist_is_on = !blacklist_is_on;

        ApplicationConfiguration.getConfiguration().setProperty(ApplicationConfiguration.BLACKLIST_IS_ON, blacklist_is_on);
        Daten.getInstance().getListeBlacklist().filterListe();
        MessageBus.getMessageBus().publishAsync(new BlacklistChangedEvent());
    }
}

File: src/main/kotlin/mediathek/gui/actions/ToggleBlacklistAction.kt
package mediathek.gui.actions

import mediathek.config.Daten
import mediathek.config.application.ApplicationConfiguration
import mediathek.gui.messages.BlacklistChangedEvent
import mediathek.swing.IconUtils
import mediathek.tool.MessageBus
import net.engio.mbassy.listener.Handler
import org.kordamp.ikonli.materialdesign2.MaterialDesignL
import org.kordamp.ikonli.swing.FontIcon
import java.awt.Color
import java.awt.event.ActionEvent
import javax.swing.AbstractAction
import javax.swing.SwingUtilities

class ToggleBlacklistAction : AbstractAction() {
    private val enabledIcon: FontIcon = IconUtils.windowBarSpecificToolbarIcon(MaterialDesignL.LIST_STATUS)
    private val disabledIcon: FontIcon = IconUtils.windowBarSpecificToolbarIcon(MaterialDesignL.LIST_STATUS, Color.RED)
    private var blacklistIsOn: Boolean = ApplicationConfiguration.getInstance().isBlacklistEnabled

    init {
        setupState()
        MessageBus.messageBus.subscribe(this)
    }

    @Suppress("UNUSED_PARAMETER")
    @Handler
    private fun handleBlacklistChangedEvent(event: BlacklistChangedEvent) {
        SwingUtilities.invokeLater {
            blacklistIsOn = ApplicationConfiguration.getInstance().isBlacklistEnabled
            setupState()
        }
    }

    private fun setupState() {
        if (blacklistIsOn) {
            putValue(NAME, "Blacklist ausschalten")
            putValue(SHORT_DESCRIPTION, "Blacklist ausschalten")
            putValue(SMALL_ICON, enabledIcon)
        } else {
            putValue(NAME, "Blacklist einschalten")
            putValue(SHORT_DESCRIPTION, "Blacklist einschalten")
            putValue(SMALL_ICON, disabledIcon)
        }
    }

    override fun actionPerformed(event: ActionEvent?) {
        blacklistIsOn = !blacklistIsOn

        ApplicationConfiguration.getInstance().isBlacklistEnabled = blacklistIsOn
        Daten.getInstance().listeBlacklist.filterListe()
        MessageBus.messageBus.publishAsync(BlacklistChangedEvent())
    }
}
